3529/1/gemhandler.rb

```diff
@@ -0,0 +1,108 @@
+require 'nokogiri'
+require 'json'
+require 'net/http'
+require 'uri'
+require 'open-uri'
+require 'httparty'
+require 'open_uri_redirections'
+require 'yaml'
+require 'octokit'
+require 'typhoeus'
+
+class GemHendler
+  attr_reader :data_about_gem
+
+  def initialize(github_url, gem_name)
+    @url = github_url
+    @data_about_gem = {}
+    puts "Didn't find repository on github" unless @url
+    @repo_addr = adress_handle
+    @client = Octokit::Client.new(login: 'gannagoodkevich', password: 'Pusivill1999')
+    @client.auto_paginate = true
+    login
+    @data_about_gem[:name] = gem_name
+    join_all_data
+  end
+
+  def login
+    user = @client.user
+    user.login
+  end
+
+  def adress_handle
+    @repo_addr = if @url.include?('https://github.com/')
+                   @url.gsub('https://github.com/', '')
+                 else
+                   @url.gsub('http://github.com/', '')
+                 end
+  end
+
+  def join_all_data
+    find_wsf
```

Comment:
What is wsf? Try to name methods / variables more clear. Code reader should get the logic what the method does from the first look